Ingress/controllers/haproxy/Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Copyright 2015 The Kubernetes Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+FROM alpine:3.2
```

Comment:
It would be nice if this was 
FROM gcr.io/google_containers/haproxy
;)

This will involve moving it to Ingress/images/haproxy, I can push it to gcr.io once you've done that. That can come as a follow up pr.